test/e2e/form-validation.test.js

```diff
@@ -10,130 +10,266 @@ describe('vl-form-validation', async () => {
     for (const uuid of invalidUuids) {
       await input.setValue(uuid);
-      await form.submit();
+      await form.submit(formNr);
       await assert.eventually.isTrue(input.hasError());
     }
 
     const validUuids = ['1c6fa548-5eef-11ea-ae93-0242ac130002', '12345678-ABCD-1234-ef00-0123456789ef'];
     for (const uuid of validUuids) {
       await input.setValue(uuid);
-      await form.submit();
+      await form.submit(formNr);
       await assert.eventually.isFalse(input.hasError());
     }
   });
 
   it('als gebruiker zie ik een foutmelding als een datum niet geldig is', async () => {
-    const form = await vlFormValidationPage.getForm(1);
+    const formNr = 1;
+    const form = await vlFormValidationPage.getForm(formNr);
     const input = await form.getInputField('datum');
 
     await input.setValue('29.02.2019');
-    await form.submit();
+    await form.submit(formNr);
     await assert.eventually.isTrue(input.hasError());
 
     await input.setValue('29.02.2020');
-    await form.submit();
+    await form.submit(formNr);
     await assert.eventually.isFalse(input.hasError());
   });
 
   it('als gebruiker zie ik een foutmelding als er niets is geselecteerd uit een lijst', async () => {
-    const form = await vlFormValidationPage.getForm(1);
+    const formNr = 1;
+    const form = await vlFormValidationPage.getForm(formNr);
     const select = await form.getSelect('stad');
 
-    await form.submit();
+    await form.submit(formNr);
     await assert.eventually.isTrue(select.hasError());
 
     await select.selectByIndex(1);
-    await form.submit();
+    await form.submit(formNr);
     await assert.eventually.isFalse(select.hasError());
   });
 
   it('als gebruiker zie ik een foutmelding als er geen datum is geselecteerd', async () => {
-    const form = await vlFormValidationPage.getForm(1);
+    const formNr = 1;
+    const form = await vlFormValidationPage.getForm(formNr);
     const datepicker = await form.getDatepicker('datum');
 
-    await form.submit();
+    await form.submit(formNr);
     await assert.eventually.isTrue(datepicker.hasError());
 
     await datepicker.selectDay(15);
-    await form.submit();
+    await form.submit(formNr);
     await assert.eventually.isFalse(datepicker.hasError());
   });
+
+  it('als gebruiker zie ik een foutmelding die gedefinieerd is in de error-placeholder als een verplicht veld niet is ingevuld', async () => {
+    const form = await vlFormValidationPage.getForm(4);
+
+    const input = await form.getInputField('form4-voornaam');
+    await assert.eventually.isTrue(input.isRequired());
+    await assert.eventually.isFalse(input.hasError());
+
+    await form.submit(4);
+    await assert.eventually.isTrue(input.hasError());
+
+    await input.setValue('Tom');
+    await form.submit(4);
+    await assert.eventually.isFalse(input.hasError());
+  });
+
+
```

Comment:
Is het nodig om elk soort validatie apart te testen? De feature "de errorboodschap kan ook hidden of unhidden gezet worden voor een input field" werkt toch hetzelfde voor alle soorten validatie?

Reply:
@coemans, mag ik alle toegevoegde testen laten staan, of hou ik er slechts 1 of enkele over. Zo ja, welke dan?

Reply:
Wat mij betreft mogen ze voorlopig blijven staan, in kader van de huidige code wijzigingen was maar één test nodig. Nu testen we naast e2e ook op unit niveau de centrale lib, wat niet nodig is maar het kan (voorlopig) ook geen kwaad.